Refactored duplicate o_parentId/parentId properties and methods

aashan10 opened the pull request:
partly fix of #8917

aashan10 committed:
Refactored duplicate o_parentId/parentId properties and methods

fashxp commented on models/Element/AbstractElement.php:
```diff
@@ -137,6 +137,40 @@ public function setPath($path)
+        $parentId = (int) $parentId;
+        $type = Service::getElementType($this);
+
+        if ($parentId != $this->parentId && $type === 'object') {
```
please avoid that kind of ifs in the parent classes ... if there is specific functionality of a subclass, then it should go into the subclass...

aashan10 replied:
Fixed by overriding the `setParentId` method in `AbstractObject` class

aashan10 committed:
Override setParentId method for DataObjects and remove element specific dependency

aashan10 committed:
Type cast parentId to integer before comparison

aashan10 committed:
Merge branch '10.x' of https://github.com/pimcore/pimcore into feature/migration/o_parentId

fashxp commented:
@aashan10  can you please merge 10.x once more?

aashan10 committed:
Merge branch '10.x' of https://github.com/pimcore/pimcore into feature/migration/o_parentId
